File: android/app/src/main/java/com/kreativekoala/scribeai/utils/TikTokHelper.kt
package com.kreativekoala.scribeai.utils

import android.content.Context
import android.util.Log
import com.tiktok.TikTokBusinessSdk

/**
 * TikTok Business SDK integration for install attribution and event tracking.
 */
object TikTokHelper {
    private const val TAG = "TikTokHelper"
    private const val APP_ID = "7606855069115449351"

    fun initialize(context: Context) {
        try {
            val config = TikTokBusinessSdk.TTConfig(context.applicationContext)
                .setAppId(APP_ID)
                .setLogLevel(TikTokBusinessSdk.LogLevel.INFO)

            TikTokBusinessSdk.initializeSdk(config)
            Log.i(TAG, "TikTok SDK initialized with appId: $APP_ID")
        } catch (e: Exception) {
            Log.e(TAG, "Failed to initialize TikTok SDK", e)
        }
    }

    fun trackEvent(eventName: String, properties: Map<String, String> = emptyMap()) {
        try {
            TikTokBusinessSdk.trackEvent(eventName)
            Log.d(TAG, "Tracked event: $eventName")
        } catch (e: Exception) {
            Log.e(TAG, "Failed to track event: $eventName", e)
        }
    }
}

File: ios/scribeai/TikTokHelper.swift
import Foundation
import TikTokBusinessSDK
import AppTrackingTransparency

/// TikTok Business SDK integration for install attribution and event tracking.
final class TikTokHelper {
    static let shared = TikTokHelper()

    private let appId = "7610310074174685191"

    private init() {}

    /// Call once at app launch.
    func initialize() {
        guard let config = TikTokConfig(appId: appId, tiktokAppId: appId) else {
            print("[TikTok] Failed to create config")
            return
        }
        config.setLogLevel(TikTokLogLevelInfo)
        TikTokBusiness.initializeSdk(config)
        print("[TikTok] SDK initialized with appId: \(appId)")
    }

    /// Request App Tracking Transparency permission for IDFA access.
    func requestTrackingPermission() {
        if #available(iOS 14, *) {
            ATTrackingManager.requestTrackingAuthorization { status in
                switch status {
                case .authorized:
                    print("[TikTok] ATT authorized — IDFA available")
                case .denied:
                    print("[TikTok] ATT denied — limited attribution")
                case .notDetermined:
                    print("[TikTok] ATT not determined")
                case .restricted:
                    print("[TikTok] ATT restricted")
                @unknown default:
                    break
                }
            }
        }
    }

    /// Track a custom event.
    func trackEvent(_ name: String, properties: [String: Any] = [:]) {
        let event = TikTokBaseEvent(eventName: name)
        for (key, value) in properties {
            event.addProperty(withKey: key, value: value)
        }
        TikTokBusiness.trackTTEvent(event)
    }
}
